policy/release/tasks.rego

```diff
@@ -0,0 +1,59 @@
+# METADATA
+# custom:
+#   tasks_required:
+#     rule_data:
+#       required_task_refs:
+#       - add-sbom-and-push
+#       - clamav-scan
+#       - deprecated-image-check
+#       - get-clair-scan
+#       - sanity-inspect-image
+#       - sanity-label-check
+#       - sanity-optional-label-check
+#       - sast-go
+package policy.release.tasks
+
+import data.lib
+import future.keywords.in
+
+# This generates all errors that can be omitted from the `tasks_required`
+# rule. Since required tasks can change over time, we need this so we
+# don't need to repeat the list of tasks in the test where this list of
+# errors is solely used. It needs to be placed here to be able to access
+# the package level metadata/annotations.
+all_tasks_missing_errors = {e | task := rego.metadata.chain()[_].annotations.custom.tasks_required.rule_data.required_task_refs[_]; e := {
+	"code": "tasks_required",
+	"msg": sprintf("Required task '%s' not found in the PipelineRun attestation", [task]),
+	"effective_on": "2022-01-01T00:00:00Z",
+}}
+
+# METADATA
+# title: No tasks run
+# description: |-
+#   This policy enforces that at least one Task is preset in the PipelineRun
+#   attestation.
+# custom:
+#   short_name: tasks_missing
+#   failure_msg: No tasks found in PipelineRun attestation
+deny[result] {
+	att := lib.pipelinerun_attestations[_]
+	count(att.predicate.buildConfig.tasks) == 0
+	result := lib.result_helper(rego.metadata.chain(), [])
+}
+
+# METADATA
+# title: Required tasks not run
+# description: |-
+#   This policy enforces that the required set of tasks is run in a
+#   PipelineRun.
+# custom:
+#   short_name: tasks_required
+#   failure_msg: Required task '%s' not found in the PipelineRun attestation
+deny[result] {
+	att := lib.pipelinerun_attestations[_]
+	attested_tasks := {t | t := att.predicate.buildConfig.tasks[_].name}
+	required_tasks := {t | t := rego.metadata.chain()[_].annotations.custom.tasks_required.rule_data.required_task_refs[_]}
```

Comment:
Maybe it's worth adding a comment here to explain the intentions since it's not 100% explicit, e.g. "# We're expecting this to find the required_task_refs list in the package level annotations, see above" or something along those lines.

Reply:
Found good use of `all_tasks_missing_errors` to simplify this and make it easier to grasp